tiledb/sm/rest/test/unit_rest_profile.cc

```diff
@@ -53,51 +53,32 @@ struct expected_values_t {
 
 struct RestProfileFx {
  public:
-  std::string homedir_;              // The user's $HOME directory.
-  TemporaryLocalDirectory tempdir_;  // A temporary working directory.
-  std::string cloudpath_;            // The in-test path to the cloud.json file.
+  std::string homedir_;    // The temporary in-test $HOME directory.
+  std::string cloudpath_;  // The in-test path to the cloud.json file.
 
-  /**
-   * Constructor which changes the user's `$HOME` to a `TemporaryLocalDirectory`
-   * for the duration of the test. This ensures that no in-test changes are
-   * made to the local, user-sensitive files `profiles.json` and `cloud.json`.
-   */
   RestProfileFx()
-      : homedir_(tiledb::common::filesystem::home_directory().value())
-      , tempdir_(TemporaryLocalDirectory("unit_rest_profile"))
-      , cloudpath_(tempdir_.path() + ".tiledb/cloud.json") {
-    if (setenv("HOME", tempdir_.path().c_str(), 1) != 0) {
-      std::cerr << "[ERROR] Failed to set HOME environment variable."
-                << std::endl;
-      return;
-    }
-
+      : homedir_(TemporaryLocalDirectory("unit_rest_profile").path())
```

Comment:
Won't this immediately delete the directory? You can instead:

* Keep `TemporaryLocalDirectory tempdir_`.
* Create a  function `RestProfile create_profile(const std::string& name = RestProfile::DEFAULT_NAME)` that passes the temporary home directory to the created profile, and update tests to call it. This will keep things simpler and reduce duplication.